raster_analysis/query_executor.py

```diff
@@ -27,9 +27,8 @@ def execute(self) -> DataFrame:
         mask = filter_mask * self.data_cube.mask
 
         if self.query.base.layer in self.data_cube.windows:
-            mask *= self.data_cube.windows[self.query.base.layer].data.astype(
-                dtype=np.bool
-            )
+            base_layer = self.query.data_environment.get_layer(self.query.base.layer)
+            mask *= self.data_cube.windows[base_layer.name].data != base_layer.no_data
```

Comment:
`nan != nan`  returns `True` so this will fail.  `np.isnan` should work though.

Reply:
Good catch, @solomon-negusse. Are we missing a pytest case that will catch this?

Reply:
There might not actually be an NaN values in the test float raster, so it's not actually removing anything. So it may require adding NaN values to a test raster.

Reply:
> There might not actually be an NaN values in the test float raster, so it's not actually removing anything. So it may require adding NaN values to a test raster.

Is that easy enough to do as part of this PR?